File: kotlin/src/test/kotlin/Bowling180702/Bowling180702Test.kt
package Bowling180702

import org.junit.Test
import kotlin.test.assertEquals

class Bowling180702Test {

    @Test
    fun testAllStrikes() {
        var b = Bowling180702()
        (1..13).forEach { b.roll(10) }
        assertEquals(300, b.score);
    }

    @Test
    fun testAllSpares() {
        var b = Bowling180702()
        (1..21).forEach { b.roll(5) }
        assertEquals(150, b.score);
    }
    
    @Test
    fun testAllNines() {
        var b = Bowling180702()
        (1..10).forEach { 
            b.roll(9)
            b.roll(0)
        }
        assertEquals(90, b.score)
    }

    @Test
    fun testAllSparesWith1stNine() {
        var b = Bowling180702()
        (1..10).forEach {
            b.roll(9)
            b.roll(1)
        }
        b.roll(9)
        assertEquals(190, b.score)
    }

}

File: cpp/test/Bowling180702Test.cpp
#include "../Bowling180702.h"
#include "gtest/gtest.h"

TEST(Bowling180702, allStrike)
{
    Bowling180702 *b = new Bowling180702();
    for (int i = 0; i < 12; i++) {
        b->roll(10);
    }
    EXPECT_EQ(300, b->score());
}

TEST(Bowling180702, allSpare)
{
    Bowling180702 *b = new Bowling180702();
    for (int i = 0; i < 21; i++) {
        b->roll(5);
    }
    EXPECT_EQ(150, b->score());
}

TEST(Bowling180702, allNine)
{
    Bowling180702 *b = new Bowling180702();
    for (int i = 0; i < 10; i++) {
        b->roll(9);
        b->roll(0);
    }
    EXPECT_EQ(90, b->score());
}

TEST(Bowling180702, allSpareWith1stNine)
{
    Bowling180702 *b = new Bowling180702();
    for (int i = 0; i < 10; i++) {
        b->roll(9);
        b->roll(1);
    }
    b->roll(9);
    EXPECT_EQ(190, b->score());
}
